src/transforms/select.jl

```diff
@@ -2,6 +2,46 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+struct TableSelection{T} 
+  table::T
+  cols::Vector{Symbol}
+  function TableSelection(table::T, cols::Vector{Symbol}) where {T}
+    @assert cols ⊆ Tables.columnnames(table)
+    new{T}(table, cols)
+  end
+end
+
+function Base.:(==)(a::TableSelection, b::TableSelection)
+  a.cols != b.cols && return false
+  all(Tables.getcolumn(a, col) == Tables.getcolumn(b, col) for col in a.cols)
+end
+
+function Base.show(io::IO, ts::TableSelection)
+  println(io, "TableSelection")
+  pretty_table(io, ts, vcrop_mode=:middle)
+end
+
+# Tables.jl interface
+Tables.istable(::Type{<:TableSelection}) = true
+Tables.columnaccess(::Type{<:TableSelection}) = true
+Tables.columns(t::TableSelection) = t
```

Comment:
I understand that the TableSelection already implements the AbstractColumns interface? That is why you applied Tables.getcolumn directly to TableSelection in the implementation ==?

Reply:
Exactly!